### ISSUE TITLE
Deprecate?

### ISSUE THREAD

**Issue by tingerrr** [tingerrr/typst-anti-matter#3]
I've used my own package successfully in a few reports for university, while using my unfinished template.

After starting work on my master's thesis and by extension the university template to turn it into a proper template package, I noticed that I once again got incorrect results. Namely, off-by-one errors, which I couldn't attribute to the metadata being shoved to the next page.

It was also trivial to create logical page numbering with a well-structured template, without `anti-matter`. So I'm wondering if this package carries its own weight, especially since Typst cannot use logical numbering for PDF-export if this package is used.

```typst
// unnumbered title page if needed
// ...

// front-matter
#set page(numbering: "I")
#counter(page).update(1)
// ...

// page counter anchor
#metadata(()) <front-matter>

// main document body
#set page(numbering: "1")
#counter(page).update(1)
// ...

// back-matter
#set page(numbering: "I")
// must take page breaks into account, may need to be offset by +1 or -1
#context counter(page).update(counter(page).at(<front-matter>).first())
// ...
```

A document like this will provide correct logical page numbering for PDF-export, as well as simple front- and back-matter numbering with almost no setup.

The only thing this package can do that this approach cannot, is getting the correct total number at each point to allow for `I / V` or `1 / 15` patterns. But this seems rare in practice.

**Comment by Fiwex** [comment]
> A document like this will provide correct logical page numbering for PDF-export, as well as simple front- and back-matter numbering with almost no setup.

I'm new to typst and making the transition from LaTeX. The method you describe seems simpler and easier to use without requiring more lines.

However, with the setup you describe above, everything is perfect, but when I want to show the `numbering` in a footer, it ignores the numbering instruction of the page function. This is what the documentation says.

So is there a simple way to achieve the same effect but inside a footer ?  Because the only one I've found is to put metadata in my code and an if/else in my footer.

**Comment by Fiwex** [comment]
> A document like this will provide correct logical page numbering for PDF-export, as well as simple front- and back-matter numbering with almost no setup.

Hi @tingerrr I'm new to typst and making the transition from LaTeX. The method you describe seems simpler and easier to use without requiring more lines.

However, with the setup you describe above, everything is perfect, but when I want to show the `numbering` in a footer, it ignores the numbering instruction of the page function. This is what the documentation says.

So is there a simple way to achieve the same effect but inside a footer ?  Because the only one I've found is to put metadata in my code and an if/else in my footer.

**Comment by florianb** [comment]
Hey @tingerrr - thanks a lot for the package. Ran exactly into the same error having off by one errors. Probably the contents don't legitimate the maintenance of a package - but probably it would be a good idea to put this into the typst docs as some kind of guide or best practises. It would be sad if the knowledgde disappeared, esp. for new typst users. :)

**Comment by tingerrr** [comment]
This could perhaps be added to the Typst Examples Book by @sitandr, although it seems their GitHub has been silent for a while :/

**Comment by sitandr** [comment]
Hello there! I'm completely fine, just don't have any active public projects now (and generally write code much less).

I would try to give some attention to the Book in next few days, maybe even contribute a bit to compiler again.

To be fair, it is hard to work on the Book because of the feeling that nobody is going to read the snippet you are writing. People who have already seen it are unlikely to return and look through old snippets for new ones. Sometimes I wonder whether it is a good idea to create some kind of [Discord server / other place] to collect and publish new snippets, or at least write code to somehow use cookies to highlight new/changed chapters to reader.

Getting back to the topic, is there anything else I should write about this snippet? I think I will put it into a new Snippets/Chapters/Numbering section, if you don't have any suggestions.

Maybe we could also add some code from the package for getting a right total number too?

**Comment by florianb** [comment]
Lots of 🫶 for maintaining the project - fair questions though. I think using a Discord-server will backlash one time. Unfortunately most search results today are polluted by generative contents. Nevertheless i think proper content will be relevant again.

I just fixed some numbering and outlining issues today, i'd be happy to lend a helping hand.

**Comment by tingerrr** [comment]
> Sometimes I wonder whether it is a good idea to create some kind of [Discord server / other place] to collect and publish new snippets, or at least write code to somehow use cookies to highlight new/changed chapters to reader.

I don't think so, one of the primary issues of the current community discord server is that it's not easily searchable. I often look for snippets when people as an FAQ in the discord server, but can't find them. Having the book which is indexable would be nicer and when I can, I link to that, tbh.

**Comment by alex-touza** [comment]
Thanks for the code snippet. I'd like to add that, as mentioned in [this support thread](https://discord.com/channels/1054443721975922748/1287834181610766438), in 0.11.x, if there is a page break before `#metadata`, it needs to be moved after the call, like this:

```typst
// ...

#pagebreak()
#outline()

// page counter anchor
#metadata(()) <frontmatter>
#pagebreak()

#set page(numbering: "1")
#counter(page).update(1)

// main document body

// ...
```
